### PR TITLE
Update RemoteServiceManager to use ServiceId instead of Id.Service

### PR DESCRIPTION
**Pull request by anwar6953**
Update RemoteServiceManager to use ServiceId instead of Id.Service, so that it makes versioned API calls.

http://builds.cask.co/browse/CDAP-DUT5039-1

### CODE REVIEW

File: cdap-integration-test/src/main/java/co/cask/cdap/test/remote/RemoteServiceManager.java [hunk 1]
```diff
@@ -42,12 +42,12 @@
   private final MetricsClient metricsClient;
   private final ProgramClient programClient;
   private final ServiceClient serviceClient;
-  private final Id.Service serviceId;
+  private final ServiceId serviceId;
 
   public RemoteServiceManager(ServiceId programId, ClientConfig clientConfig, RESTClient restClient,
```
Comment (ecfm) [inline cdap-integration-test/src/main/java/co/cask/cdap/test/remote/RemoteServiceManager.java]: Rename `programId` to `serviceId`?

**Comment by ecfm** [comment]
You can also update the `getServiceURL(ProgramId service)` method in `ServiceClient` to accept `ServiceId` now.

**Comment by ecfm** [comment]
lgtm

**Comment by anwar6953** [comment]
build is green; squashing and merging.

**Pull request merged**